src/contract/assignments.rs

```diff
@@ -150,6 +151,25 @@ impl ToMerkleSource for OwnedRightsInner {
     }
 }
 
+impl MergeRevealed for OwnedRights {
+    fn merge_revealed(&mut self, other: &Self) -> Result<bool, String> {
```

Comment:
See my comment in the general review on using `String` types instead of `Error` and on returning `Result`

Reply:
Noted: I used String for initial chalking out of the trait. Updating with dedicated error type in next push.